### PR TITLE
Python: Fix the link in the README

### PR DESCRIPTION
**Pull request by Fokko**

### CODE REVIEW

File: python/README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 # Iceberg Python
 
-pyiceberg is a python library for programmatic access to iceberg table metadata as well as to table data in iceberg format. It is a Python implementation of [iceberg table spec](https://iceberg.apache.org/spec/). Documentation is available at [https://pyiceberg.apache.org/](https://pyiceberg.apache.org/).
+pyiceberg is a python library for programmatic access to iceberg table metadata as well as to table data in iceberg format. It is a Python implementation of [iceberg table spec](https://iceberg.apache.org/spec/). Documentation is available at [https://py.iceberg.apache.org/](https://py.iceberg.apache.org/).
```
Comment (ajantha-bhat) [inline python/README.md:20]: I know that this was the working link. But when I click now, I get a 404 error. Can you please check once?

Comment (Fokko) [reply python/README.md:20]: Thanks for reporting. I also noticed this yesterday and pinged Apache INFRA: https://issues.apache.org/jira/browse/INFRA-23715

Comment (Fokko) [reply python/README.md:20]: Also created a new ticket: https://issues.apache.org/jira/browse/INFRA-23738

Comment (Fokko) [reply python/README.md:20]: It works again, thanks again @ajantha-bhat for pointing out 👍🏻

**Review by ajantha-bhat — APPROVED** [review]

**Review by nastra — APPROVED** [review]

**Pull request merged**